### PR TITLE
[AIRFLOW-803] Revert join with dag_runs in _execute_task_instances

### PR DESCRIPTION
**Pull request by bolkedebruin**
TaskInstances will be set to 'scheduled' if they meet the criteria to run,
 also the ones up for retry. No task_instance will be send to the executor
in another state than 'scheduled'.

Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- AIRFLOW-803

@jhsenjaliya @aoen @plypaul

### CODE REVIEW

File: airflow/jobs.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 from time import sleep
```
Comment (aoen) [inline airflow/jobs.py]: We probably want to remove the DEADLOCK_CHECK_DEP_CONTEXT logic that we added along with this.

**Comment by aoen** [comment]
LGTM

**Comment by alexvanboxel** [comment]
+1

**Comment by jhsenjaliya** [comment]
+1 LGTM, I have tested this and I no longer see the issue i had, Thanks !

**Comment by plypaul** [comment]
IIRC filtering based on the DAG run was to handle an edge case. I can't recall the exact scenario, but one that I just thought up was that a DAG run and the associated tasks are created by the scheduler, but then before the tasks are able to run, the user deletes the DAG run. Arguably, then such tasks shouldn't be sent to the executor.

**Pull request merged**